cmd/rqlite/execute.go

```diff
@@ -25,96 +26,70 @@ type executeResponse struct {
 	Time    float64   `json:"time,omitempty"`
 }
 
-func executeWithClient(ctx *cli.Context, client *http.Client, argv *argT, timer bool, stmt string) error {
+func executeWithClient(ctx *cli.Context, client *cl.Client, timer bool, stmt string) error {
 	queryStr := url.Values{}
 	if timer {
 		queryStr.Set("timings", "")
 	}
 	u := url.URL{
-		Scheme: argv.Protocol,
-		Host:   fmt.Sprintf("%s:%d", argv.Host, argv.Port),
-		Path:   fmt.Sprintf("%sdb/execute", argv.Prefix),
+		Path: fmt.Sprintf("%sdb/execute", client.Prefix),
 	}
-	urlStr := u.String()
 
 	requestData := strings.NewReader(makeJSONBody(stmt))
 
-	nRedirect := 0
-	for {
-		if _, err := requestData.Seek(0, io.SeekStart); err != nil {
-			return err
-		}
-
-		req, err := http.NewRequest("POST", urlStr, requestData)
-		if err != nil {
-			return err
-		}
-		if argv.Credentials != "" {
-			creds := strings.Split(argv.Credentials, ":")
-			if len(creds) != 2 {
-				return fmt.Errorf("invalid Basic Auth credentials format")
-			}
-			req.SetBasicAuth(creds[0], creds[1])
-		}
+	if _, err := requestData.Seek(0, io.SeekStart); err != nil {
+		return err
+	}
 
-		resp, err := client.Do(req)
-		if err != nil {
-			return err
-		}
-		response, err := ioutil.ReadAll(resp.Body)
-		if err != nil {
-			return err
-		}
-		resp.Body.Close()
+	resp, err := client.Execute(u, requestData)
 
-		if resp.StatusCode == http.StatusUnauthorized {
-			return fmt.Errorf("unauthorized")
-		}
+	if err != nil {
+		return err
+	}
+	response, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	resp.Body.Close()
 
-		if resp.StatusCode == http.StatusMovedPermanently {
-			nRedirect++
-			if nRedirect > maxRedirect {
-				return fmt.Errorf("maximum leader redirect limit exceeded")
-			}
-			urlStr = resp.Header["Location"][0]
-			continue
-		}
+	if resp.StatusCode == http.StatusUnauthorized {
```

Comment:
Do you think this check adds anything, versus the `resp.StatusCode != http.StatusOK` immediately afterwards? I don't think it does, and the code is cleaner without it.

Reply:
I left it there in case you want to add a specific handler for that specific case, otherwise, if we are just logging back errors, it does not add anything.

If you are not planning on dealing with 403 errors separately, I can remove the check.

Reply:
No plans right now to deal with them separately, I say remove.

Reply:
Done!